src/main/java/org/powertac/logtool/LogtoolCore.java

```diff
@@ -61,66 +72,130 @@ public LogtoolCore ()
+  {
+    if (null == source || "-".equals(source)) {
+      log.info("Reading from standard input");
+      return readStateLog(System.in, tools);
+    }
+    try {
+      URL inputURL = new URL(source);
+      log.info("Reading url " + source);
+      return readStateLog(inputURL,tools);
+    } catch (MalformedURLException x) {
+      // Continue, assuming it is a regular file
+    }
+    log.info("Reading file " + source);
+    File inputFile = new File(source);
+    if (!inputFile.canRead()) {
+      return "Cannot read file " + source;
+    }
+    return readStateLog(inputFile, tools);
+  }
+
+  /**
+   * Reads state-log from given input file using the DomainObjectReader.
+   */
+  public String readStateLog (File inputFile, Analyzer... tools)
+  {
+    try{
+      return readStateLog(new FileInputStream(inputFile), tools);
+    } catch (FileNotFoundException e) {
+      return "Cannot open file " + inputFile.getPath();
+    }
+  }
+
+  /**
+   * Reads state-log from given input url using the DomainObjectReader.
+   */
+  public String readStateLog (URL inputURL, Analyzer... tools)
+  {
+    try{
+      return readStateLog(inputURL.openStream(), tools);
+    } catch (IOException e) {
+      return "Cannot open url " + inputURL.toString();
+    }
   }
 
   /**
-   * Reads the given state-log file using the DomainObjectReader.
-   * If a filename is given, we assume it names a state log file.
-   * If filename is "-" or null, then we assume the state log is
-   * on standard-input. This allows for piping the input from
-   * a compressed archive.
+   * Reads state-log from given input stream using the DomainObjectReader.
    */
-  public void readStateLog (String filename, List<Analyzer> tools)
+  public String readStateLog (InputStream inputStream, Analyzer... tools)
   {
-    reader.registerNewObjectListener(new SimEndHandler(),
-                                     SimEnd.class);
+    reader.registerNewObjectListener(new SimEndHandler(), SimEnd.class);
     Reader inputReader;
     String line = null;
+
     try {
-      if (null == filename || "-".equals(filename)) {
-        // read from standard input
-        inputReader = new InputStreamReader(System.in);
-        filename = "standard input";
-        log.info("Reading from standard input");
+      // Stack compression logic if appropriate
+      try {
+        if (!inputStream.markSupported()) {
+          inputStream = new BufferedInputStream(inputStream);
+        }
+        inputStream = compressFactory.createCompressorInputStream(inputStream);
+      } catch (CompressorException x) {
+        // Stream not compressed (or unknown compression scheme)
       }
-      else {
-        // explicit filename given
-        File input = new File(filename);
-        if (!input.canRead()) {
-          System.out.println("Cannot read file " + filename);
-          return;
+
+      // Stack archive logic if appropriate
+      try {
+        if (!inputStream.markSupported()) {
```

Comment:
Yeah, I thought so too. But unfortunately the `CompressorInputStream` that might have been wrapped earlier doesn't support marks. I found that odd as well, since this combination of both compression and archiving seems common enough to me.